### ISSUE TITLE
[Auction Setup] Create Auction Reward

### ISSUE THREAD

**Issue by clatwell** [Good-Bids/goodbids#422]
This ticket details what we need to do to for Step 2 (Create Auction Reward) of the Auction Guided Setup, along with some open questions for Nick and Brian. I've split it out into four sections to keep details organized: 

- See reward form 
- Add data to form 
- Submit form 
- Store product id   

### Admins see Create Auction Reward Form in Auction Setup Wizard 

![Image](https://github.com/Good-Bids/goodbids/assets/16839493/90a0f61b-3def-41ba-afee-bc5d834cfd0f)

Here are some things we'll need to do to be able to display the form: 

**Fetch Shipping Classes for Nonprofit** 

Physical Rewards will (optionally) have a Shipping Class associated with them. We need to be able to show Admins a list of the Shipping Classes that the NP has created for their WC store. We should be able to retrieve a [list of all shipping classes](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-shipping-classes) with a GET request to ` /wp-json/wc/v3/products/shipping_classes` before the form renders, so we can list Shipping Classes (by name) in the dropdown shown below. 

**Hook up WordPress File Uploader** 

When an Admin creates a Product in the WooCommerce UI, they have the option to 1) add one Product Image, then 2) add multiple Product Gallery images. We'd like to extend this capability to Admins creating Reward products in the Setup flow. _I think_ we'll be doing ourselves a favor here if we can hook into the existing WP File Uploader / Media Library UI for this. Brian noted that he has a chunk of code I can use to pull in the Media Library and add a separate method to implement it. 

An additional nice-to-have here would be to hook into the WP UI that displays uploaded images, so that we can show them on the form page. We can track this has an enhancement if we don't get it "for free" with the above. 

![Image](https://github.com/Good-Bids/goodbids/assets/16839493/0af4b51b-6b1b-474d-a6cc-b0a54845d863)

**Show Tooltips** 

I am 97% confident that we will need to include tooltips on this form page. I don't know which specific form fields or form section headers will need tooltips, or what the content in the tooltips would be. I'm recommending that we go ahead and build a pattern to support tooltips (open on click or hover, show plain text) so that we can come in and refine them later. If this seems better-suited to track in an enhancement ticket, lmk and we can remove from this first pass.

### Admins can enter data in the Create Auction Reward Form 

- **Reward Name** 
  - Input type: String 
  - Tooltip: Yes 
  - Required: Yes
- **Fair Market Value** 
  - Input type: Number / Currency (value with one monetary decimal point without thousand separators or current symbols) 
  - Tooltip: Yes 
  - Required: Yes 
- **Set featured image** 
  - Opens WP File Upload / Media Library overlay 
  - Admins can select up to one file 
  - Required: No 
- **Add product gallery images** 
  - Opens WP File Upload / Media Library overlay 
  - Admins can select many files 
  - Required: No 
- **What type of product is it?** 
 
For this input, Admins will see radio buttons that allow them to select one Reward Type (Physical, Virtual, Experience). Based on their selection... 

- The form field(s) below the radio button input will adjust 
- The Product created upon form submission will have varying fields / attributes, detailed below

For Physical products, Admins should see fields for Weight, Dimensions, and Shipping Class. All of these are optional. All will have tooltips. 

For Virtual and Experience products, Admins should see a text field for "Redemption Instructions." This field _is_ required (if the previous selection != Physical) and will have a tooltips. 

| Syntax      | Description |
| ----------- | ----------- |
| ![Image](https://github.com/Good-Bids/goodbids/assets/16839493/2ea57385-830d-4247-8e49-8ec63fa96d82)  | ![Image](https://github.com/Good-Bids/goodbids/assets/16839493/67eecfb9-28a1-448e-9b2d-828fbef9d7ab)       |

### Admins can submit the form to Create a Reward Product 

**Post to WooCommerce API** 

When the Admin submits the form, assuming all required fields are present and have valid data, we'll want to [create a product](https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-product) by sending a POST request to ` /wp-json/wc/v3/products`. 

In the body of the POST request, we'll need to pass through attributes that the Admin entered in the form, attributes that correspond to the defaults we will set for all products, and attributes that map to the Reward Type inputs. 

**Admin-entered Attributes** 

- Reward => `name` 
- Fair Market Value => `regular-price` 
- Images => An array of `images` with the `src` URL 
  - Note: It's not obvious to me in the API docs that there is a way programmatic way to indicate which of the images in the array is the featured image. I assume WC is treating the first image in the array as the featured image. We should validate that assumption. 

**Default Attributes** 

- `type` => `simple` 
- `catalog_visibility` => `hidden`
- `categories` - We need to add the Reward category to each product created in this form. The POST request body can include an array of categories by id, so we need to pass the ID of the Reward category. We will need to fetch the id of the reward category for the site: 
  - get_category_id(): [Source](https://github.com/Good-Bids/goodbids/blob/main/client-mu-plugins/goodbids/src/classes/Auctions/Rewards.php#L78) | [Docs](https://github.com/Good-Bids/goodbids/blob/main/docs/api/rewards-methods.md?plain=1#L8)
- `manage_stock` => `true` 
- `stock_quantity` => `1` 
- `sold_individually` => `true`
- `reviews_allowed` => `false`

**Mapped Attributes** 

Note: Working on reward type documentation in parallel, will link that up shortly, I suspect we may actually want to add a custom product field to track which of these the admin selected. 

- Physical Product 
  - `tax_status` => `taxable` 
  - `tax_rate` => `standard` (Note that we're still finalizing all the sales tax details) 
  - `weight`=> user input, optional
  - `dimensions` => user input, array, see [here](https://woocommerce.github.io/woocommerce-rest-api-docs/#product-dimensions-properties), optional 
  - `shipping_class` => slug of shipping class user selected, optional 
- Virtual Product 
  - `virtual` => `true`
  - `tax_status` => `standard`
  - `tax_rate` => `zero_rate` (Note that we're still finalizing all the sales tax details)   
  - `purchase_note` => user input in "reward redemption instructions" field 
- Experience Product 
  - `virtual` => `true`
  - `tax_status` => `none` 

### The Auction Setup gets back the id of the product 

Once the reward is successfully created via the WooCommerce API, the API will respond with the `id` of the product that was just created. We need to carry that `id` through to the next page (Step 3: Add Auction Details) so the product with the Auction once it gets created.
